lamindb/models/query_set.py

```diff
@@ -921,6 +922,44 @@ def get(self, idlike: int | str | None = None, **expressions) -> SQLRecord:
 
     def filter(self, *queries, **expressions) -> QuerySet:
         """Query a set of records."""
+        if expressions and self.model.__name__ == "Artifact":
+            from ._feature_manager import filter_base
+            from .artifact import Artifact
+            from .feature import Feature
+
+            keys_normalized = [key.split("__")[0] for key in expressions]
+            field_or_feature = keys_normalized[0].split("__")[0]
+            available_fields = {f.name for f in Artifact._meta.get_fields()}
```

Comment:
No, this doesn't help, the problem is from BaseQuerySet.filter called via super, not from QuerySet.filter, so this doesn't have any effect at all. I don't think this is needed at all.

Reply:
This is needed to decide whether we have a feature-based or a Django-based query. And yes, it fixed the initial issues in this PR.